src/EditorFeatures/CSharpTest/Completion/CompletionProviders/SpeculativeTCompletionProviderTests.cs

```diff
@@ -162,6 +162,474 @@ class C
             await VerifyItemExistsAsync(markup, "T");
         }
 
+        [Fact, Trait(Traits.Feature, Traits.Features.Completion)]
+        [WorkItem(37224, "https://github.com/dotnet/roslyn/issues/37224")]
+        public async Task InRef()
+        {
+            var markup = @"
+using System;
+class C
+{
+    ref $$
```

Comment:
Does this also need the `ref T$$` case?

Reply:
I think `T$$` cases are very important as we are about to hit <kbd>space</kbd> or other commit characters.

Reply:
Added.